src/Lykke.Service.RateCalculator.Services/RateCalculatorService.cs

```diff
@@ -85,18 +85,20 @@ public double GetRate(string neededAssetId, AssetPair assetPair, double price)
             var marketProfile = await GetMarketProfileRemoteAsync();
             var pricesData = _crossPairsCalculator.PrepareForConversion(marketProfile);
 
-            foreach (var chunk in balanceRecords.Batch(10))
+            if (balanceRecords != null &&
+                balanceRecords.Any())
```

Comment:
balanceRecords is enumerated twice but it is an IEnumerable. Either change IEnumerable to IReadOnlyCollection or remove double enumeration of the balanceRecords